src/index.tsx

```diff
@@ -23,22 +23,124 @@ async function main(GitURL: string, branchName?: string, folderPath?: string) {
     await $`git checkout ${branchName}`;
 
     await fs.remove(path.join(tempFolder, '.git'));
-    await fs.copy(
-        folderPath ? path.join(tempFolder, folderPath) : tempFolder,
-        targetFolder,
-        { overwrite: true }
-    );
+    
+    const sourceFolder = folderPath ? path.join(tempFolder, folderPath) : tempFolder;
+    
+    // Check if source is a file or directory
+    const sourceStat = await fs.stat(sourceFolder);
+    
+    if (sourceStat.isFile()) {
+        // If it's a file, copy it directly to target directory with same name
+        const fileName = path.basename(sourceFolder);
+        await fs.copy(sourceFolder, path.join(targetFolder, fileName), { overwrite: true });
+    } else {
+        // If it's a directory, copy contents to target directory
+        await fs.copy(sourceFolder, targetFolder, { overwrite: true });
+    }
+}
+
+async function removeSubmodule(submodulePath?: string) {
+    const currentDir = process.cwd();
+    
+    // If no path provided, list current submodules
+    if (!submodulePath) {
+        try {
+            const result = await $`git submodule status`;
+            console.log('Current submodules:');
+            console.log(result.stdout);
+            console.log('\nUsage: xgit submodule remove <path>');
+            return;
+        } catch (error) {
+            console.error('Error: This does not appear to be a git repository with submodules.');
+            process.exit(1);
+        }
+    }
```

Comment:
```suggestion
    if (!submodulePath) {
            const result = await $`git submodule status`;
            console.log('Current submodules:');
            console.log(result.stdout);
            console.log('\nUsage: xgit submodule remove <path>');
            return;
    }
```
1. 原始命令行的错误直接自然抛出
2. 直接用多行字符串来一次性输出
3. 这段缺参数的逻辑单独抽一个函数